message_ix/macro.py

```diff
@@ -416,10 +417,18 @@ def add_model_data(base, clone, data):
     for n in c.nodes:
         clone.add_set("cat_node", ["economy", n])
 
-    # add sectoral set structure
-    for s, l in product(c.sectors, c.levels):
+    # Obtain the mapping data frame inlcuding sector, level and commodity
+    sector_mapping_df = c.sector_mapping
+
+    # Add sectoral set structure
+    # Itearte throguh rows in the mapping data frame
+
+    for i in sector_mapping_df.index:
+        s = sector_mapping_df.iloc[i, sector_mapping_df.columns.get_loc("sector")]
+        c = sector_mapping_df.iloc[i, sector_mapping_df.columns.get_loc("commodity")]
```

Comment:
Thanks @GamzeUnlu95 . Did you test the code after adding your changes? I get an error, which seems to be due to reintroducing variable `c` in the code. This variable is once reserved for `Calculate`. So, maybe you name "commodity" as "com" in the for loop.